Docs: Add note about exception for integer slices with float indices (#58017)
pandas-dev/pandas: doc/source/user_guide/indexing.rst

described in the :ref:`Selection by Position <indexing.integer>` section
detailing the ``.iloc`` method. For now, we explain the semantics of slicing using the ``[]`` operator.

>     .. note::
>         
>         When the :class:`Series` has float indices, slicing will select by position.

Reviewer: ```suggestion
    .. note::

        When the :class:`Series` has float indices, slicing will select by position.
```

described in the :ref:`Selection by Position <indexing.integer>` section
detailing the ``.iloc`` method. For now, we explain the semantics of slicing using the ``[]`` operator.

> However, there is an exception to this rule when the Series has float indices.
> In such cases, integer slices behave more the ``.iloc`` method

Reviewer: ```suggestion
    .. note::
        
        When the :class:`Series` has float indices, slicing will select by position. 
```
